common/persistence/client/bean.go

```diff
@@ -181,6 +191,13 @@ func (s *BeanImpl) GetExecutionManager() persistence.ExecutionManager {
 	return s.executionManager
 }
 
+// GetNexusServiceManager get NexusServiceManager
+func (s *BeanImpl) GetNexusServiceManager() persistence.NexusServiceManager {
```

Comment:
A bit curious why this needs to be a "bean".
Can't we inject this everywhere it's needed?

Reply:
That's a good question, I'm not 100% sure. My guess is it is to group all the persistence dependencies together so if someone wants to replace them with a different implementation, they only need to replace one provider.